### PR TITLE
Fix duplicate mob squash logic in 3D tutorial (Your first 3D game) 

### PR DESCRIPTION
**Pull request by jjaxu**
fixes https://github.com/godotengine/godot-docs/issues/7592

Hi there! 👋 

I was following the [Godot 3D tutorial (Your first 3D game)](https://docs.godotengine.org/en/stable/getting_started/first_3d_game/index.html) and believe to have discovered a bug regarding squashing the mobs.

The [collision detection](https://docs.godotengine.org/en/stable/getting_started/first_3d_game/06.jump_and_squash.html#coding-the-squash-mechanic) section mentions `"Godot makes the body move sometimes multiple times in a row to smooth out the character's motion. So we have to loop over all collisions that may have happened."` and thus checks for collisions in a loop.

This current logic however will sometimes result in duplicate calls to `mob.squash()`, which will double or triple count the score for one mob kill. The simple fix is to `break` out of the loop after handling the first squash occurrence to prevent further duplicate calls.

### Demo
Here's the demo, notice how the first two kills increased the score by two, then the last two by one. I've seen anywhere between 1-3 
demo

I've updated all occurrences of this code snippet that appear for both GDScript and C# to include the `break` statement, as well as an explanation of the logic. The files changed are:
- 06.jump_and_squash
- 07.killing_player 
- 09.adding_animations

### Preview
GDScript
image
C#
image
Explanation
image

Please help verify whether this is the correct solution. If it is, I'm also more than happy to apply the fix to the [godot-3d-dodge-the-creeps](https://github.com/godotengine/godot-3d-dodge-the-creeps/blob/master/dodge_the_creeps_3d/Player.gd#L60) repo as well.

Thanks and have a great day!

* *Bugsquad edit, alternative to: #8333*

### CODE REVIEW

File: getting_started/first_3d_game/06.jump_and_squash.rst
Comment (AThousandShips) [inline getting_started/first_3d_game/06.jump_and_squash.rst:258]:
```suggestion
                   # Prevent further duplicate calls.
```
End comments with a period, same elsewhere

**Comment by jjaxu** [comment]
I believe this fixes: https://github.com/godotengine/godot-docs/issues/7592

**Comment by AThousandShips** [comment]
To link it add the line "Fixes: #7592" to your first message (the one at the very top)

**Comment by jjaxu** [comment]
Linked the issue. Glad to see using `break` was the intended solution. What about the standalone [godot-3d-dodge-the-creeps](https://github.com/godotengine/godot-3d-dodge-the-creeps/blob/master/dodge_the_creeps_3d/Player.gd#L60) repo? Seems to me that it's no longer maintained and the demos have since been migrated to [godot-demo-projects](https://github.com/godotengine/godot-demo-projects/blob/master/3d/squash_the_creeps)

**Comment by AThousandShips** [comment]
That does seem to be the case yes

**Comment by jjaxu** [comment]
@AThousandShips Is there anything else that needs to be done before this can be approved and merged? There is nothing else from my side.

**Comment by AThousandShips** [comment]
You'll have to see what further reviewers say

**Comment by sdlllllll** [comment]
I'm a beginner and also noticed this bug in the tutorial. Is that mean if I use collision detection in any calculation, a "break" or "return" is necessary in order to prevent a duplicate calculation.

**Comment by george-lim** [comment]
This would be a very helpful for onboarding new people, I also ran into this bug.

@AThousandShips any chance we could get another pair of eyes from the team to review? It's definitely a nice to have and the changes seem harmless / trivial :)

**Review by AThousandShips — APPROVED** [review]
Looks good to me, haven't verified the behavior but is the same logic as the 2D case

**Comment by jjaxu** [comment]
Thanks @AThousandShips, I have squashed my commits and rebased the changes onto the latest `master`.

**Comment by jjaxu** [comment]
@AThousandShips  Commit https://github.com/godotengine/godot-docs/commit/426c83b added some whitespace changes which caused merge conflicts. I've rebased my changes again to resolve. Will require approval again. Thanks!

**Review by Calinou — APPROVED** [review]
Thanks!

**Comment by J-John1** [comment]
Thanks for your help everyone!

**Pull request merged**
